pytensor/tensor/basic.py

```diff
@@ -2758,29 +2758,21 @@ def concatenate(tensor_list, axis=0):
     return join(axis, *tensor_list)
 
 
-def horizontal_stack(*args):
+def hstack(*args):
     r"""Stack arrays in sequence horizontally (column wise)."""
-    # Note: 'horizontal_stack' and 'vertical_stack' do not behave exactly like
-    # Numpy's hstack and vstack functions. This is intended, because Numpy's
-    # functions have potentially confusing/incoherent behavior (try them on 1D
-    # arrays). If this is fixed in a future version of Numpy, it may be worth
-    # trying to get closer to Numpy's way of doing things. In the meantime,
-    # better keep different names to emphasize the implementation divergences.
 
     if len(args) < 2:
         raise ValueError("Too few arguments")
 
     _args = []
     for arg in args:
         _arg = as_tensor_variable(arg)
-        if _arg.type.ndim != 2:
```

Comment:
Can you add a test for the special behavior with 1d inputs (and mixed 1d and 2d) to make sure it matches numpy?